bleep-model/src/scala/bleep/model/ScriptDef.scala

```diff
@@ -1,24 +1,24 @@
 package bleep.model
 
-import io.circe.generic.semiauto.deriveCodec
+import bleep.RelPath
 import io.circe._
-import io.circe.syntax.EncoderOps
+import io.circe.generic.semiauto.deriveCodec
 
 sealed trait ScriptDef {
-  def sourceGlobs: Option[SourcesGlobs]
+  lazy val asJson: Json = ScriptDef.encodes(this)
```

Comment:
out of curiosity, any reason why you don't like the `def sourceGlobs` at the sealed-trait level ?

Reply:
Normally there is absolutely nothing wrong with that. In this particular case, as of right now, the two `sourceGlobs` do *not* mean the same thing.

```
            script match {
              case model.ScriptDef.Main(project, _, sourceGlobs) =>
                val projectDir = buildPaths.project(project, build.explodedProjects(project)).dir
                sourceGlobs.values.toList.map(relPath => Config.SourcesGlobs(projectDir / relPath, None, Nil, Nil))
              case model.ScriptDef.Shell(_, _, sourceGlobs) =>
                val projectDir = buildPaths.project(crossName, explodedProject).dir
                sourceGlobs.values.toList.map(relPath => Config.SourcesGlobs(projectDir / relPath, None, Nil, Nil))
            }
          }
```

If you look carefully, the relative paths are added to two different absolute paths, one is relative to the script project and one relative to the project which receives the sources. This is what I talked about in one of the comments above. So no usage of the shared `sourceGlobs` would be correct. That's why I removed it

Reply:
Oh, gotcha 👍